pkgs/development/python-modules/conda/default.nix

```diff
@@ -1,38 +1,77 @@
 { lib
 , buildPythonPackage
-, pythonAtLeast
-, fetchPypi
+, pythonRelaxDepsHook
+, fetchFromGitHub
+, hatchling
+, hatch-vcs
+, archspec
+, conda-libmamba-solver
+, conda-package-handling
+, distro
+, jsonpatch
+, packaging
+, platformdirs
+, pluggy
 , pycosat
 , requests
 , ruamel-yaml
-, isPy3k
-, enum34
+, tqdm
+, truststore
 }:
-
 # Note: this installs conda as a library. The application cannot be used.
 # This is likely therefore NOT what you're looking for.
-
 buildPythonPackage rec {
   pname = "conda";
-  version = "4.3.16";
-  format = "setuptools";
-
-  # this is a very outdated version of conda that isn't compatible with python 3.10+
-  disabled = pythonAtLeast "3.10";
+  version = "24.1.2";
+  pyproject = true;
 
-  src = fetchPypi {
+  src = fetchFromGitHub {
     inherit pname version;
-    sha256 = "a91ef821343dea3ba9670f3d10b36c1ace4f4c36d70c175d8fc8886e94285953";
+    owner = "conda";
+    repo = "conda";
+    rev = version;
+    hash = "sha256-L/Y7Bb3R5YqXbjTN4CRPFnkgymVLrxuFmjVzpvt28dE=";
   };
 
-  propagatedBuildInputs = [ pycosat requests ruamel-yaml ] ++ lib.optional (!isPy3k) enum34;
+  buildInputs = [ hatchling hatch-vcs ];
+  propagatedBuildInputs = [
+    pythonRelaxDepsHook
+    archspec
+    conda-libmamba-solver
+    conda-package-handling
+    distro
+    jsonpatch
+    packaging
+    platformdirs
+    pluggy
+    pycosat
+    requests
+    ruamel-yaml
+    tqdm
+    truststore
+  ];
+
+  patches = [ ./0001-conda_exe.patch ];
+  # menuinst is a dependency that is only needed on Windows
+
+  # default path to store conda environments
+  CONDA_ENVS_PATH = "~/.conda/envs";
+  # default path to store download conda packages
+  CONDA_PKGS_DIRS = "~/.conda/pkgs";
```

Comment:
Are they only used for `makeWrapperArgs` or is it also necessary to pass them as environment variables to the builder? In the former case, please move to a let-in expression